dccrg.hpp

```diff
@@ -9891,6 +9903,12 @@ template <
 			this->all_to_all_set(new_donts);
 		} while (new_donts.size() > 0);
 
+		for (const auto& cell : old_donts) {
+			if(!this->is_local(cell)) {
+				old_donts.erase(cell);
+			}
+		}
```

Comment:
And this is the one that ended up growing so much.

so the above loop continues until it performs a cycle after which all processes return `new_donts.size()==0`.

It starts off with the (local) information of `this->cells_not_to_refine`, and any induced non-refinement tags go into new_donts which are all-to-alled, so at the end of this looping, each task should have both its original donts, and all induced donts from the whole grid. Thus, before doing the below all-to-all, we indeed want to remove all non-local donts, and end up with the total as intended. So I think this is safe.